scripts/missions/windurst/1_2_The_Heart_of_the_Matter.lua

```diff
@@ -0,0 +1,477 @@
+        {
+            ['Chawo_Shipeynyo'] = mission:event(110),
+            ['Keo-Koruo']       = mission:event(108),
+            ['Pakke-Pokke']     = mission:event(109),
+            ['Zokima-Rokima']   = mission:event(107),
+        },
+
+        [xi.zone.WINDURST_WATERS] =
+        {
+            ['Dagoza-Beruza'] = mission:event(132),
+            ['Mokyokyo']      = mission:event(131),
+            ['Panna-Donna']   = mission:event(133),
+            ['Ten_of_Hearts'] = mission:event(134),
+        },
+
+        [xi.zone.WINDURST_WOODS] =
+        {
+            ['Miiri-Wohri'] = mission:event(133),
+            ['Rakoh_Buuma'] = mission:event(136),
+            ['Sola_Jaab']   = mission:event(134),
+            ['Tih_Pikeh']   = mission:event(135),
+        },
+    },
+
+    -- Head to the Manustery located in Windurst Woods (Home Point #1) to speak with Apururu at (H-9) who will give
+    -- you 6 Dark Mana Orbs and tell you to go to the Outer Horutoto Ruins.
+    {
+        check = function(player, currentMission, missionStatus)
+            return currentMission == mission.missionId and missionStatus == 1
+        end,
+
+        [xi.zone.WINDURST_WOODS] =
+        {
+            ['Apururu'] = mission:progressEvent(137),
+
+            onEventFinish =
+            {
+                [137] = function(player, csid, option, npc)
+                    npcUtil.giveKeyItem(player, {
+                        xi.ki.FIRST_DARK_MANA_ORB,
+                        xi.ki.SECOND_DARK_MANA_ORB,
+                        xi.ki.THIRD_DARK_MANA_ORB,
+                        xi.ki.FOURTH_DARK_MANA_ORB,
+                        xi.ki.FIFTH_DARK_MANA_ORB,
+                        xi.ki.SIXTH_DARK_MANA_ORB
+                    })
+                    player:setMissionStatus(mission.areaId, 2)
+                end,
+            },
+        },
+    },
+
+    -- Head to East Sarutabaruta and go to the Marguerite Tower at (J-11); speak with the Tarutaru named Pore-Ohre
+    -- and he will give you a Key Item Southeastern Star Charm.
+    {
+        check = function(player, currentMission, missionStatus)
+            return currentMission == mission.missionId and missionStatus == 2
+        end,
+
+        [xi.zone.WINDURST_WOODS] =
+        {
+            ['Apururu'] = mission:progressEvent(138),
+        },
+
+        [xi.zone.EAST_SARUTABARUTA] =
+        {
+            ['Pore-Ohre'] = mission:progressEvent(46),
+
+            onEventFinish =
+            {
+                [46] = function(player, csid, option, npc)
+                    npcUtil.giveKeyItem(player, xi.ki.SOUTHEASTERN_STAR_CHARM)
+                    player:setMissionStatus(mission.areaId, 3)
+                end,
+            },
+        },
+    },
+
+    -- Handle both the placing and later collecting of mana orbs here
+    {
+        check = function(player, currentMission, missionStatus)
+            return currentMission == mission.missionId and missionStatus == 3 or missionStatus == 4
```

Comment:
Not entirely clear the order of operations you want here, please add parens to clarify